lib/db_connection/connection_pool.ex

```diff
@@ -358,4 +372,9 @@ defmodule DBConnection.ConnectionPool do
     idle = :erlang.start_timer(timeout, self(), now_in_native, abs: true)
     %{codel | idle: idle}
   end
+
+  defp decrement_or_zero_active_conn(metrics) do
```

Comment:
Why do we need this logic? If at some point we can go to -1, it means at some point we were off by one?

Reply:
Our main usage of this was Xandra, and we noticed that when Xandra started up the connection pool, it seems to checkin the connections, so a simple -1 each time would cause your pool metrics to startup as `{-N, 0}` and the highest point it would reach was 0.